### PR TITLE
New Unit LMTank and fix of #227

### PR DESCRIPTION
**Pull request by LMOliver**
The new unit `LMTank` shoots a `EnergyBeam` every frame when firing.

The `EnergyBeam` is a bullet that exists for 1 frame, and hits the nearest unit or obstacle on its direction immediately. It has a maximum range of 10.

### CODE REVIEW

**Comment by LMOliver** [comment]
It also closes #227.

File: src/battle_game/core/units/tiny_tank.cpp [hunk 1]
```diff
@@ -131,9 +131,7 @@ void Tank::TurretRotate() {
 }
 
 void Tank::Fire() {
-  if (fire_count_down_) {
-    fire_count_down_--;
-  } else {
+  if (fire_count_down_ == 0) {
```
Comment (LazyJazz) [inline src/battle_game/core/units/tiny_tank.cpp:134]: Undo this.

**Review by LazyJazz — COMMENTED** [review]
LGTM! Just undo the unnecessary change.

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**